genesyscloud/resource_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,218 @@
+package genesyscloud
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/leekchan/timeutil"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v53/platformclientv2"
+)
+
+func getAllArchitectSchedules(ctx context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(ResourceIDMetaMap)
+	archAPI := platformclientv2.NewArchitectApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		schedules, _, getErr := archAPI.GetArchitectSchedules(pageNum, 100, "", "", "", nil)
+		if getErr != nil {
+			return nil, diag.Errorf("Failed to get page of schedules: %v", getErr)
+		}
+
+		if schedules.Entities == nil || len(*schedules.Entities) == 0 {
+			break
+		}
+
+		for _, schedule := range *schedules.Entities {
+			resources[*schedule.Id] = &ResourceMeta{Name: *schedule.Name}
+		}
+	}
+
+	return resources, nil
+}
+
+func architectSchedulesExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllArchitectSchedules),
+		RefAttrs:         map[string]*RefAttrSettings{}, // No references
+	}
+}
+
+func resourceArchitectSchedules() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Architect Schedules",
+
+		CreateContext: createWithPooledClient(createArchitectSchedules),
+		ReadContext:   readWithPooledClient(readArchitectSchedules),
+		UpdateContext: updateWithPooledClient(updateArchitectSchedules),
+		DeleteContext: deleteWithPooledClient(deleteArchitectSchedules),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Name of the schedule.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the schedule.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+            "start": {
```

Comment:
formatting is off here. You can use go format to format the file. https://go.dev/blog/gofmt

Reply:
Also, it would be good for consistency to create a function like validateDate from validators.go to validate LocalDateTimes and use it as a validation function for `start` and `end`